platform/security/rest/security-rest-cxfwrapper/src/test/java/org/codice/ddf/cxf/paos/PaosInInterceptorTest.java

```diff
@@ -0,0 +1,167 @@
+import ddf.security.liberty.paos.impl.RequestBuilder;
+import ddf.security.liberty.paos.impl.RequestMarshaller;
+import ddf.security.liberty.paos.impl.RequestUnmarshaller;
+import ddf.security.liberty.paos.impl.ResponseBuilder;
+import ddf.security.liberty.paos.impl.ResponseMarshaller;
+import ddf.security.liberty.paos.impl.ResponseUnmarshaller;
+
+public class PaosInInterceptorTest {
+
+    @Before
+    public void setup() {
+        OpenSAMLUtil.initSamlEngine();
+        XMLObjectProviderRegistry xmlObjectProviderRegistry = ConfigurationService.get(
+                XMLObjectProviderRegistry.class);
+        xmlObjectProviderRegistry.registerObjectProvider(Request.DEFAULT_ELEMENT_NAME,
+                new RequestBuilder(), new RequestMarshaller(), new RequestUnmarshaller());
+        xmlObjectProviderRegistry.registerObjectProvider(Response.DEFAULT_ELEMENT_NAME,
+                new ResponseBuilder(), new ResponseMarshaller(), new ResponseUnmarshaller());
+    }
+
+    @Test
+    public void handleMessagePaosResponseBasicGood() throws IOException {
+        Message message = new MessageImpl();
+        message.setContent(InputStream.class, PaosInInterceptorTest.class.getClassLoader()
+                .getResource("ecprequest.xml")
+                .openStream());
+        message.put(Message.CONTENT_TYPE, "application/vnd.paos+xml");
+        Message outMessage = new MessageImpl();
+        HashMap<String, List> protocolHeaders = new HashMap<>();
+        outMessage.put(Message.PROTOCOL_HEADERS, protocolHeaders);
+        protocolHeaders.put("Authorization", Collections.singletonList("BASIC dGVzdDp0ZXN0"));
+        ExchangeImpl exchange = new ExchangeImpl();
+        exchange.setOutMessage(outMessage);
+        message.setExchange(exchange);
+        PaosInInterceptor paosInInterceptor = new PaosInInterceptor(Phase.RECEIVE) {
+            HttpResponseWrapper getHttpResponse(String url, String soapResponse)
+                    throws IOException {
+                HttpResponseWrapper httpResponseWrapper = new HttpResponseWrapper();
+                if (url.equals("https://sp.example.org/PAOSConsumer")) {
+                    httpResponseWrapper.statusCode = 200;
+                    httpResponseWrapper.content = new ByteArrayInputStream(
+                            "actual content".getBytes());
+                } else if (url.equals("https://idp.example.org/saml2/sso")) {
+                    httpResponseWrapper.statusCode = 200;
+                    httpResponseWrapper.content = PaosInInterceptorTest.class.getClassLoader()
+                            .getResource("idpresponse.xml")
+                            .openStream();
+                }
+                return httpResponseWrapper;
+            }
+        };
+        paosInInterceptor.handleMessage(message);
+        assertThat(IOUtils.toString(message.getContent(InputStream.class)), is("actual content"));
+    }
+
+    @Test(expected = Fault.class)
```

Comment:
It looks like there are more fault states we could unit tests for this interceptor.